### PR TITLE
16 improve gradle build

### PR DESCRIPTION
**Pull request by slifty**
This PR represents progress towards #16.  It removes ant from the project and fully replaces it with gradle.

Internal dependencies are now much more clear within the build script, and models are rebuilt as part of the standard build script.

### CODE REVIEW

File: psm-app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        mavenCentral()
+    }
+}
+
+project(":services") {
+    apply plugin: 'java'
+
+    dependencies {
+        compile fileTree(dir: '../cms-portal-services/lib', include: '*.jar', exclude: 'cms-core.jar')
+	compile project(path: ":cms-business-model", configuration: 'archives')
```
Comment (jasonaowen) [inline psm-app/build.gradle]: Indent and quotes should be consistent with the other dependencies.

File: psm-app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            }
+        }
+    }
+}
+
+project(":cms-business-process") {
+    apply plugin: 'java'
+    dependencies {
+        runtime project(path: ":services", configuration: 'archives')
+        compile project(path: ":services", configuration: 'archives')
```
Comment (jasonaowen) [inline psm-app/build.gradle]: I don't think this needs both `runtime` and `compile`; [`runtime` includes everything in `compile`](https://docs.gradle.org/3.5/userguide/userguide_single.html#configurations).

File: psm-app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            generatePackage = 'gov.medicaid.domain.model'
+        }
+    }
+}
+
+project(":cms-portal-services") {
+    apply plugin: 'ear'
+    dependencies {
+        deploy project(path: ":cms-web", configuration: 'archives')
+	deploy project(":cms-business-process")
```
Comment (jasonaowen) [inline psm-app/build.gradle]: Indent, quotes.

File: psm-app/build.xml [hunk 4]
```diff
@@ -1,197 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<project name="CMS Portal Services" default="dist" basedir=".">
```
Comment (jasonaowen) [inline psm-app/build.xml]: Woo bye bye ant!

File: psm-app/cms-business-model/src/main/resources/jaxb-bindings.xjb [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <jaxb:bindings version="2.1" xmlns:jaxb="http://java.sun.com/xml/ns/jaxb" xmlns:xjc="http://java.sun.com/xml/ns/jaxb/xjc" xmlns:xs="http://www.w3.org/2001/XMLSchema">
-	<jaxb:globalBindings generateElementProperty="false">
+        <jaxb:globalBindings generateElementProperty="false">
```
Comment (jasonaowen) [inline psm-app/cms-business-model/src/main/resources/jaxb-bindings.xjb]: Why this whitespace change?

File: psm-app/gradle.properties [hunk 6]
```diff
@@ -0,0 +1 @@
+wildfly.home=../../wildfly-10.1.0.Final
```
Comment (jasonaowen) [inline psm-app/gradle.properties]: Should we templatize this, as we did for `build.properties`? And, speaking of which, should we remove `build.properties.template` in this PR?

File: psm-app/services/src/main/java/gov/medicaid/dao/impl/LDAPIdentityProviderDAOBean.java [hunk 7]
```diff
@@ -1,456 +0,0 @@
...
-/**
- * This is an EJB implementation for the {@link IdentityProviderDAO} which connects directly to configurable LDAP
- * servers.
- *
- * @author TCSASSEMBLER
- * @version 1.0
- */
-@Stateless
-@Local(IdentityProviderDAO.class)
-public class LDAPIdentityProviderDAOBean implements IdentityProviderDAO {
```
Comment (jasonaowen) [inline psm-app/services/src/main/java/gov/medicaid/dao/impl/LDAPIdentityProviderDAOBean.java]: Why did you remove this file?

File: psm-app/build.gradle [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    }
+    jaxb {
+        xsdDir = "cms-business-model/src/main/resources"
+        bindingsDir = "cms-business-model/src/main/resources"
+        bindings = ["jaxb-bindings.xjb"]
+        xjc {
+            producesDir = 'src/main/generated'
+            destinationDir = 'src/main/generated'
+            generatePackage = 'gov.medicaid.domain.model'
+        }
```
Comment (jasonaowen) [inline psm-app/build.gradle]: It's so cool you figured out how to do this elegantly in gradle!

**Comment by jasonaowen** [comment]
Ahhh this is awesome I can't wait to have this

**Comment by cecilia-donnelly** [comment]
By the way, @slifty, sorry for late response but: :confetti_ball: :100: :1st_place_medal: !!

File: psm-app/gradle.properties (hunk 6 above)
Comment (slifty) [inline psm-app/gradle.properties]: It should be (this slipped by my gitignore).
That said, it isn't currently used, and once we refactor dependency management it is possible we won't need a path to wildfly to build in the first place.

File: psm-app/build.gradle (hunk 8 above)
Comment (slifty) [inline psm-app/build.gradle]: This was a fun one to research for sure ;)

File: psm-app/cms-business-model/src/main/resources/jaxb-bindings.xjb (hunk 5 above)
Comment (slifty) [inline psm-app/cms-business-model/src/main/resources/jaxb-bindings.xjb]: Total accident during an experiment; good catch.

File: psm-app/services/src/main/java/gov/medicaid/dao/impl/LDAPIdentityProviderDAOBean.java (hunk 7 above)
Comment (slifty) [inline psm-app/services/src/main/java/gov/medicaid/dao/impl/LDAPIdentityProviderDAOBean.java]: Early in the process the file was giving me some guff and removing the file fixed an error -- at this point it is clear that error was a red herring, however; adding the file back now.

**Comment by cecilia-donnelly** [comment]
Thanks @slifty and @jasonaowen!
